File: frontend/src/types/Video.ts
// Data coming from the backend
export type VideoResponse = {
  id: number;
  contractId: number;
  accountId: number;
  platformVideoId: string;
  platformVideoLink?: string;
  location?: string;
  title?: string;
  description?: string;
  duration?: string;
  postedAt?: string;
  isOnTime?: string;
  createdAt: string;
};

// Data sent when syncing/creating a video
export type VideoSyncRequest = {
  contractId: number;
  accountId: number;
  platformVideoId: string;
  platformVideoLink?: string;
  location?: string;
  title?: string;
  description?: string;
  duration?: string;
  postedAt?: string;
};

File: backend/src/main/groovy/com/krasen/vizor/business/domain/Video.java
package com.krasen.vizor.business.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.OffsetDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Video {
    private Long id;
    private Contract contract;
    private Account account;
    private String platformVideoId;
    private String platformVideoLink;
    private String location;
    private String title;
    private String description;
    private String duration;
    private String postedAt;
    private String isOnTime;
    private OffsetDateTime createdAt;
    private OffsetDateTime deletedAt;
}
